service/controller/v21/templates/cloudformation/guest/security_groups.go

```diff
@@ -143,4 +143,22 @@ const SecurityGroups = `{{define "security_groups" }}
       ToPort: -1
       SourceSecurityGroupId: !Ref MasterSecurityGroup
 
+  VPCDefaultSecurityGroupIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId !GetAtt VPC.DefaultSecurityGroup
+      IpProtocol: -1
+      FromPort: -1
+      ToPort: -1
+      SourceSecurityGroupId: !Ref VPC.DefaultSecurityGroup
```

Comment:
I would definitely add `CidrSource: "127.0.0.1/32"`, same for the egress.
Also adding `Description` would be beneficial as it will also tell customers a reason why this is locked. 
`Description: "Lockdown default security Group."`

Reply:
After some research it makes sense to add the loopback addr for the egress rule as stated by @calvix  . Regarding the ingress rule I'd say is ok to work like that. Please advice if you don't agree with that.

Reference:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html#w2ab1c21c10d105d104c15b4

Reply:
Its indeed redundant, yet it's much more easy to read :)

also `!Ref VPC.DefaultSecurityGroup` this might not work as there is no such resource named `VPC.DefaultSecurityGroup` in CloudFormation template so Reference that might not work.

Reply:
`VPC.DefaultSecurityGroup` it should work. `VPC` is the name of block which reference the VPC object. `DefaultSecurityGroup` is a special property name in CloudFormation as here described https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html#w2ab1c21c10d105d174c11b4

Reply:
Oh special property, Nice.

Reply:
I agree with the readability however I don't suggest to use the loopback address for the ingress rule because for the fact that in case that two or more EC2 instances have assigned the default security group then they can't communicate anymore between them. This security ingress rule instead is a little bit less restricted for instances inside the VPC but at the same time very safe for whatever thing which run outside.